LiteDB/Utils/Extensions/WaitHandleExtensions.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace LiteDB.Utils.Extensions
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    internal static class WaitHandleExtensions
+    {
+        public static async Task<bool> WaitAsync(this WaitHandle handle)
+        {
+            var tcs = new TaskCompletionSource<bool>();
+
+            using (new ThreadPoolRegistration(handle, tcs))
+            {
+                return await tcs.Task.ConfigureAwait(false);
+            }
+        }
+
+        private sealed class ThreadPoolRegistration : IDisposable
+        {
+            private readonly RegisteredWaitHandle _registeredWaitHandle;
+
+            public ThreadPoolRegistration(WaitHandle handle, TaskCompletionSource<bool> tcs)
+            {
+                _registeredWaitHandle = ThreadPool.RegisterWaitForSingleObject(handle,
+                    (state, timedOut) => ((TaskCompletionSource<bool>)state).TrySetResult(!timedOut), tcs,
+                    Timeout.InfiniteTimeSpan, executeOnlyOnce: true);
```

Comment:
do we want the Timeout to be infinite? I suggest using a large time, for example 1 day. You would be surprise for systems that get deadlocked because of infinite time spans

Reply:
I can't agree here. What if there is an app that for some reason does not write anything for 2 days? I don't think we can pick any reasonable timeout here. 

Regarding deadlocks - we explicitly call Set() in DiskWriterQueue.Dispose() method, so even with infinite timeout we won't have deadlocks here.